app/components/additional-fields-input.js

```diff
@@ -44,5 +44,9 @@ export default Ember.TextField.extend({
 
   click() {
     this.addCssStyle();
+  },
+
+  willDestroyElement() {
+    this.destroy();
```

Comment:
Interesting 🤔 
Curious to know, Why should `this.destroy` be called explicitly? 

Isnt it the case that when component is being removed from DOM, the `willDestroyElement` is invoked?
So the point that its invoking `willDestroyElement` is it will be executing `this.destroy()` next.

In short, `willDestroyElement ` is a clean up method hook before `this.destroy` is invoked
Isn't that right?

Reply:
In ember we have to explicitly destroy the component. When we were doing browser back button, the component was still in the DOM giving reference error, also it is better to destroy the component if you are leaving the page to avoid memory leaks.

Reply:
It does feel weird

Reply:
What was the original bug behind this fix ?

Reply:
We are doing ajax request on focusOut in component and when were going out of this page with browser back at that time  ajax was in process to cancel it, i had to explicity destroy the element.

Reply:
I don't think this fix is right, 

but maybe you can try something like

```javascript
  focusOut() {
    if (this.get('isDestroying') || this.get('isDestroyed')) return;

    const config = {
      value: this.get("value"),
      name: this.get("name"),
      previousValue: this.get("previousValue")
    };
    this.get("onFocusOut")(config);
    this.set("inlineTextInput", false);
  },
```

ref: https://api.emberjs.com/ember/3.21/classes/Component/properties/isDestroying?anchor=isDestroying

Reply:
Sorry for dropping unannounced but component will get destroy once route is changed on clicking back link not before that so it won't prevent component from sending AJAX request. 
(Assuming the case is clicking browser back while still editing the field)

I think a good solution to this would be checking if event => target object has any DOM elements while focusing out/clicking on something else

```
focusOut(event) {

}
```

CC: @abulsayyad123